from testapp.models import ExampleModel

from djangosanetesting.cases import DatabaseTestCase, DestructiveDatabaseTestCase
 
class TestDatabaseRollbackCase(DatabaseTestCase):
    """
    Check we got proper rollback when trying to play with models.
    """
    def test_inserting_two(self):
        # guard assertion
        self.assert_equals(0, len(ExampleModel.objects.all()))
        ExampleModel.objects.create(name="test1")
        ExampleModel.objects.create(name="test2")

        # check we got stored properly
        self.assert_equals(2, len(ExampleModel.objects.all()))

    def test_inserting_two_again(self):
        # guard assertion, will fail if previous not rolled back properly
        self.assert_equals(0, len(ExampleModel.objects.all()))

        ExampleModel.objects.create(name="test3")
        ExampleModel.objects.create(name="test4")

        # check we got stored properly
        self.assert_equals(2, len(ExampleModel.objects.all()))

class TestFixturesLoadedProperly(DestructiveDatabaseTestCase):
    fixtures = ["random_model_for_testing"]
    
    def test_model_loaded(self):
        self.assert_equals(2, len(ExampleModel.objects.all()))